File: backend/mr/src/main/java/io/featurehub/mr/resources/oauth2/providers/GithubEmail.java
package io.featurehub.mr.resources.oauth2.providers;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

@JsonIgnoreProperties(ignoreUnknown = true)
public class GithubEmail {
  @JsonProperty("email")
  private String email;
  @JsonProperty("primary")
  private Boolean primary;
  @JsonProperty("verified")
  private Boolean verified;
  @JsonProperty("visibility")
  private String visibility;

  public String getEmail() {
    return email;
  }

  public void setEmail(String email) {
    this.email = email;
  }

  public Boolean getPrimary() {
    return primary;
  }

  public void setPrimary(Boolean primary) {
    this.primary = primary;
  }

  public Boolean getVerified() {
    return verified;
  }

  public void setVerified(Boolean verified) {
    this.verified = verified;
  }

  public String getVisibility() {
    return visibility;
  }

  public void setVisibility(String visibility) {
    this.visibility = visibility;
  }
}

File: backend/security-oauth/src/main/kotlin/io/featurehub/web/security/oauth/providers/GithubEmail.kt
package io.featurehub.web.security.oauth.providers

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
class GithubEmail {
  @JsonProperty("email")
  var email: String? = null

  @JsonProperty("primary")
  var primary: Boolean? = null

  @JsonProperty("verified")
  var verified: Boolean? = null

  @JsonProperty("visibility")
  var visibility: String? = null
}
